core/meta/model.js

```diff
@@ -64,17 +64,23 @@ var Model = exports.Model = Montage.specialize( /** @lends Model.prototype # */
     deserializeSelf: {
         value: function (deserializer) {
             var value = deserializer.getProperty("version");
-            if (value !== undefined) {
+            if (value !== void 0) {
                 this.version = value;
             }
 
-            this._name = deserializer.getProperty("name");
+            value = deserializer.getProperty("name");
+            if (value !== void 0) {
+                this._name = value;
+            }
             //copy contents into the objectDescriptors array
             value = deserializer.getProperty("objectDescriptors") || deserializer.getProperty("blueprints");
             if (value) {
                 this._objectDescriptors = value;
             }
-            this.modelInstanceModuleId = deserializer.getProperty("objectModelModuleId") || deserializer.getProperty("binderModuleId");
+            value = deserializer.getProperty("objectModelModuleId") || deserializer.getProperty("binderModuleId");
```

Comment:
It would be a good idea to add comments to explain that now a deserialization can be re-entring which is why the deserialization logic has to be adapted. This is something we should document and explain as well

Reply:
Agreed, but deserializeSelf is not a concretely defined api on a specific type. The deserializer just looks for a `deserializeSelf` method on whatever object it loads, and then expects it to work the way it wants. `deserializeSelf` isn't specific to the Montage type so I'm not sure where we would comment this in the code.
For general documentation, we should probably have some doc on the website that explains the serialization process and how to hook into it using serializeSelf and deserializeSelf. I think that would be good to have but outside the scope of this PR.

Reply:
Agreed, we'll need to write a technical doc